tljh_opendatacube.py

```diff
@@ -42,28 +50,54 @@ def setup_default_products():
 
 def setup_database_for_datacube():
 
+    logger.info('Setting up the Postgresql DB for the Open Data Cube...')
+    # TODO: create a read-only account for non-admin users
+
+    # enable then restart teh service
     sh.systemctl("enable", "postgresql")
     sh.service("postgresql", "restart")
 
-    su_postgres = sh.su.bake("-", "postgres", "-c")
+    su_postgres = sh.su.bake("-", "postgres", "-c")  # preamble
 
     # ensure we're starting from scratch
-    su_postgres(f"psql -c \'DROP DATABASE IF EXISTS {DATABASE_NAME};\'")
+    su_postgres(f"psql -c \'DROP DATABASE IF EXISTS {ODC_DB_NAME};\'")
     su_postgres("psql -c \'DROP EXTENSION IF EXISTS postgis;\'")
-    su_postgres(f"psql -c \'DROP ROLE IF EXISTS {DB_ADMIN_ROLE};\'")
-    su_postgres(f"psql -c \'DROP ROLE IF EXISTS {DB_USER_ROLE};\'")
+    su_postgres(f"psql -c \'DROP ROLE IF EXISTS {ODC_DB_ADMIN_USER};\'")
+    su_postgres(f"psql -c \'DROP ROLE IF EXISTS {ODC_DB_READ_ONLY_USER};\'")
 
     # create database + extension
     su_postgres("psql -c \'CREATE EXTENSION postgis;\'")
-    su_postgres(f"psql -c \'CREATE DATABASE {DATABASE_NAME};\'")
+    su_postgres(f"psql -c \'CREATE DATABASE {ODC_DB_NAME};\'")
+
+    # set a pw for the postgres db user
+    su_postgres(f"psql -c \"ALTER USER {POSTGRES_DB_USER} PASSWORD \'{POSTGRES_DB_PASS}\';\"")
+
+    # initialize the datacube
+    su_postgres(f"source /opt/tljh/user/bin/activate && DB_HOSTNAME={PSQL_HOST} DB_USERNAME={POSTGRES_DB_USER} DB_PASSWORD={POSTGRES_DB_PASS} DB_DATABASE={ODC_DB_NAME} datacube -v system init")
 
-    # configure postgres to work with datacube
-    su_postgres(f"psql -c \"ALTER USER postgres PASSWORD \'{POSTGRES_PW}\';\"")
-    su_postgres(f"source /opt/tljh/user/bin/activate && DB_HOSTNAME=localhost DB_USERNAME=postgres DB_PASSWORD={POSTGRES_PW} DB_DATABASE={DATABASE_NAME} {DATABASE_NAME} -v system init")
-    su_postgres(f"psql -c \"CREATE ROLE {DB_ADMIN_ROLE} WITH LOGIN IN ROLE {ODC_ADMIN_ROLE}, {ODC_USER_ROLE} ENCRYPTED PASSWORD \'{DB_ADMIN_PW}\';\"")
-    su_postgres(f"psql -c \"CREATE ROLE {DB_USER_ROLE} WITH LOGIN IN ROLE {ODC_USER_ROLE} ENCRYPTED PASSWORD \'{DB_USER_PW}\';\"")
-    su_postgres(f"psql -c \'ALTER DATABASE {DATABASE_NAME} OWNER TO {DB_ADMIN_ROLE};\'")
+    # create an admin role in the odc db
+    su_postgres(f"psql -c \"CREATE ROLE {ODC_DB_ADMIN_USER} WITH LOGIN IN ROLE agdc_admin, agdc_user ENCRYPTED PASSWORD \'{ODC_DB_ADMIN_PASS}\';\"")
 
+    # create user role in the odc db
+    su_postgres(f"psql -c \"CREATE ROLE {ODC_DB_READ_ONLY_USER} WITH LOGIN IN ROLE agdc_user ENCRYPTED PASSWORD \'{ODC_DB_READ_ONLY_PASS}\';\"")
+    su_postgres(f"psql -c \'ALTER DATABASE {ODC_DB_NAME} OWNER TO {ODC_DB_ADMIN_USER};\'")
+
+
+def setup_odc_gee():
```

Comment:
This could probably be handled in pip since just passes to pip iirc but this is good for now